src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -141,13 +134,20 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
       const root : FiberRoot = (container.stateNode : any);
       const current = root.current;
 
-      root.pendingContext = getContextForSubtree(parentComponent);
+      const context = getContextForSubtree(parentComponent);
+      if (root.context === null) {
+        root.context = context;
+      } else {
+        root.pendingContext = context;
+      }
 
       scheduleTopLevelUpdate(current, element, callback);
 
       if (__DEV__) {
         if (ReactFiberInstrumentation.debugTool) {
-          if (element === null) {
+          if (current.type === null) {
```

Comment:
How about `current.child === null`?

I will also move this above the scheduleTopLevelUpdate because that sounds closer to how it works with non-synchronous priority and if we don't then onUpdateContainer is called before onMountContainer due to the reentrancy (the update is called from the mount and finishes before the mount does)